index.js

```diff
@@ -458,50 +487,57 @@ XiaomiRoborockVacuum.prototype = {
-                if (that.pause) {
-                    that.pausepossible = true;
-                    that.lastrobotpausecleaning = that.pausepossible;
-                    that.pauseService.getCharacteristic(Characteristic.On).updateValue(that.pausepossible);
-                    log.info('ACT setCleaning - pause | ' + that.model + ' | PausePossible set to %s.', that.pausepossible);
-                }
-            } else {
-               log.debug('DEB setCleaning | ' + that.model + ' | Start cleaning not necessary, cleaning already running.');
-            }
-
-        } else {
-            log.info('ACT setCleaning | ' + that.model + ' | Stop cleaning and go to charge.');
-            that.device.activateCharging(); // Charging works for 1st, not for 2nd
-            that.cleaning = false
-            that.lastrobotcleaning = that.cleaning;
-
-            // Cleaning stoped => pausing not possible
-            if (that.pause) {
-                that.pausepossible = false
-                that.lastrobotpausecleaning = that.pausepossible;
-                that.pauseService.getCharacteristic(Characteristic.On).updateValue(that.pausepossible);
+                return that.device.activateCleaning()
+                    // IS THIS REALLY NEEDED?
+                    // .then(() => {
+                        // that.cleaning = true
+                        // that.lastrobotcleaning = that.cleaning;
+                        // that.charging = false;
+                        // that.lastrobotcharging = that.charging;
+                        // that.batteryService.getCharacteristic(Characteristic.ChargingState).updateValue(Characteristic.ChargingState.NOT_CHARGING);
+
+                        // Cleaning => leaves dock
+                        // if (that.dock) {
+                            // that.docked = false;
+                            // that.dockService.getCharacteristic(Characteristic.OccupancyDetected).updateValue(that.docked);
+                            // log.info('ACT setCleaning - dock | ' + that.model + ' | Docked set to %s.', that.docked);
+                        // }
+
+                        // Cleaning => pausing possible
+                        // if (that.pause) {
+                        //     that.pausepossible = true;
+                        //     that.lastrobotpausecleaning = that.pausepossible;
+                        //     that.pauseService.getCharacteristic(Characteristic.On).updateValue(that.pausepossible);
+                        //     log.info('ACT setCleaning - pause | ' + that.model + ' | PausePossible set to %s.', that.pausepossible);
+                        // }
+                    // });
+            } else if (!state) { // Stop cleaning
+                log.info('ACT setCleaning | ' + that.model + ' | Stop cleaning and go to charge.');
+                return that.device.activateCharging() // Charging works for 1st, not for 2nd
+                // IS THIS REALLY NEEDED?
```

Comment:
I think so. So the status in Homekit is set and we don't wait for the "changed" via the API.